src/SocketModeReceiver.ts

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable @typescript-eslint/explicit-member-accessibility, @typescript-eslint/strict-boolean-expressions */
+import { SocketModeClient } from '@slack/socket-mode';
+import { createServer } from 'http';
+import { Logger, ConsoleLogger, LogLevel } from '@slack/logger';
+import { InstallProvider, CallbackOptions, InstallProviderOptions, InstallURLOptions } from '@slack/oauth';
+import App from './App';
+import { Receiver, ReceiverEvent } from './types';
+
+// TODO: we throw away the key names for endpoints, so maybe we should use this interface. is it better for migrations?
+// if that's the reason, let's document that with a comment.
+export interface SocketModeReceiverOptions {
+  logger?: Logger;
+  logLevel?: LogLevel;
+  clientId?: string;
+  clientSecret?: string;
+  stateSecret?: InstallProviderOptions['stateSecret']; // required when using default stateStore
+  installationStore?: InstallProviderOptions['installationStore']; // default MemoryInstallationStore
+  scopes?: InstallURLOptions['scopes'];
+  installerOptions?: InstallerOptions;
+  appToken?: string; // App Level Token
```

Comment:
Could appToken be optional here?
```suggestion
  appToken: string; // App Level Token
```

Reply:
Nice catch!